internal/zinc-core/src/main/scala/sbt/internal/inc/IncrementalCompile.scala

```diff
@@ -381,12 +384,9 @@ private final class AnalysisCallback(
   def getOrNil[A, B](m: collection.Map[A, Seq[B]], a: A): Seq[B] = m.get(a).toList.flatten
   def addCompilation(base: Analysis): Analysis =
     base.copy(compilations = base.compilations.add(compilation))
-  def addUsedNames(base: Analysis): Analysis = usedNames.foldLeft(base) {
-    case (a, (className, names)) =>
-      import scala.collection.JavaConverters._
-      names.asScala.foldLeft(a) {
-        case (a, name) => a.copy(relations = a.relations.addUsedName(className, name))
-      }
+  def addUsedNames(base: Analysis): Analysis = {
+    assert(base.relations.names.size == 0)
```

Comment:
Why?  And should we keep it?

Reply:
I added a fast path to `Relations.addUsedNames` when `this.names` is empty, as is always the case in the way Zinc is implemented. I want to test that assumption here. Maybe its worth keeping to defend the assumption.